### PR TITLE
#240 Create a Helm Chart

### PR DESCRIPTION
**Pull request by tico24**
Hopefully I've put everything in the right place. Apologies if not... it's very early here!

This is the initial attempt at a Helm chart for Sorry Cypress. By default there's no persistence, it uses the dummy screenshots driver and the inmemory db driver.

However, tweaking the values.yml file will allow the user to both create (and use) a persistent mongo instance, and configure S3 screenshot storage.

### CODE REVIEW

**Comment by agoldis** [comment]
@tico24 Sorry for taking so long. I have been afk a lot during the holidays, I appreciate a lot what you've done! Will review soon!

**Review by agoldis — APPROVED** [review]

**Comment by agoldis** [comment]
@tico24 @fsmaia I was sure that adding org members would grant sufficient permissions to create branches etc... apparently not. I have invited you to a team that has maintainer access to this repo, and admin for https://github.com/sorry-cypress/charts

Please tell me if that's sufficient

**Comment by tico24** [comment]
ok @agoldis @fsmaia I'm now not sure why the linter is still failing. I'm reluctant to merge this in with it failing.  It passes on my downstream fork:

https://github.com/tico24/sorry-cypress/runs/1654760212?check_suite_focus=true
https://github.com/tico24/sorry-cypress/runs/1654760232?check_suite_focus=true

**Comment by tico24** [comment]
Well one of the reasons was I forgot the release.yml so it never released the helm. I have addressed this now, but this hasn't fixed the issue.

**Comment by agoldis** [comment]
@tico24  I am a complete noob, but I think until the files are actually merged into master, they won't be accessible?

**Comment by tico24** [comment]
> @tico24 I am a complete noob, but I think until the files are actually merged into master, they won't be accessible?

This would make a lot of sense. Shall we just try a merge?

**Comment by agoldis** [comment]
Hm... what is `chart-testing` supposed to do? Does it test the candidate?

**Comment by tico24** [comment]
> Hm... what is `chart-testing` supposed to do? Does it test the candidate?

I would hope so, yes!

**Comment by agoldis** [comment]
> > Hm... what is `chart-testing` supposed to do? Does it test the candidate?
> 
> I would hope so, yes!

In that case probably using the PR branch is correct, otherwise it will test the version on master?

**Comment by agoldis** [comment]
sorry if it doesn't make sense - that's all quite new to me :)

**Comment by tico24** [comment]
> sorry if it doesn't make sense - that's all quite new to me :)

No, that would make sense. It would explain why it works on my fork. I don't know how to tell it to look at the PR branch though.... frantically googling :)

**Comment by tico24** [comment]
I could well be wrong, but I think it IS checking the PR code. It looks like it saying "as long as the PR is a merge to master, then run this". It then pulls the PR, merges with master and runs the tests... that's what the github action logs imply anyway.

**Comment by tico24** [comment]
I'm afraid I'm out of ideas!

File: .github/workflows/lint-test.yml
Comment (agoldis) [inline .github/workflows/lint-test.yml:42]:
```suggestion
          config: charts/lint.yaml
```

File: charts/test.yaml
Comment (agoldis) [inline charts/test.yaml:1]: is it really used?

File: .github/workflows/lint-test.yml
Comment (agoldis) [inline .github/workflows/lint-test.yml:51]: should we use an explicit config file with `debug: true`?

**Comment by agoldis** [comment]
> I'm afraid I'm out of ideas!

@fsmaia do you have any ideas?

File: charts/test.yaml
Comment (tico24) [reply charts/test.yaml:1]: it would appear not. It's also not where we're failing though.

**Comment by tico24** [comment]
I think I've worked it out. We have to release the chart first, which means it needs to be on master.. and then it should work, no? No matter what I do on my repo, it works.... so I'm still pretty confident its working. Worst case, we revert

**Comment by agoldis** [comment]
I did something here https://github.com/sorry-cypress/sorry-cypress/pull/254 and it worked, but I have no idea why it worked

**Comment by tico24** [comment]
it worked because I changed the "on: pull_request" line to "on: push".

**Pull request merged**
